### DISCUSSION TITLE
lower music volume for tts instead of pausing it

### DISCUSSION THREAD

**Discussion by scheusal333** [music-assistant/support#998] [Feature requests and ideas]
It would be nice to have the option for tts to be played "over" the music, so keep it playing in the background with lower volume instead of pause and resume. For me this would sound much smoother without interruption.

**Comment by lbschenkel** [comment]
Seconding this. I didn't find this one and created a different one that got closed as a duplicate. For visibility, I'm going to repeat what I said there (a bit more focused on ESPHome / Sendspin but applies to anything supporting queue flow):

---

When using the regular `esphome` and `assist_satellite` integrations in Home Assistant, announcements via `assist_satellite.announce` or indirectly via `tts.speak` and other ways result in the device ducking (not pausing) the audio currently playing and playing the announcement on top of it. This indicates that the ESPHome firmware has native capability to mix two audio streams at different volumes.

I checked the source in HA and ESPHome side, there's an `announcement` flag that can can be set in requests done via ESPHome API and triggers this behaviour in the device (instead of replacing the current playing stream).

Now, Music Assistant on the other hand always pauses the currently playing stream, plays the announcement, then resumes. It won't duck as it happens in the pure HA workflow, even though MA has the concept of "native announcements". 

I suspect that the issue here (and potential difficulty) may be because MA is using Sendspin, and then one (or both) of the following holds:
1. Sendspin player in ESPHome is different/complicated enough that the current implementation is not able to mix the incoming announcement with the Sendspin audio
2. Mixing may actually be possible, but since it's using the in-device logic and not Sendspin player for the announcement it means that the announcement audio will not be synchronized with the same guarantees that Sendspin provides

Regardless of reason, can't ducking be implemented by mixing the audio in Music Assistant instead? Sendspin already works in "queue flow" mode anyway, Music Assistant could duck the playing stream and mix it with a parallel one with the announcement. This would work with any other provider that operates (or can operate) in queue flow mode: Snapcast, Squeezelite, etc. Everything for that should be in place already since cross-fading songs is already implemented and it's basically the same thing, just not at the end of the a song.

Thoughts?

**Comment by OzGav** [comment]
This is available for player types that support native announcements which currently is Sonos and Sendspin. As Sendspin now has bridges to many other protocols you should find this is available for most of your players if you use Sendspin as the default protocol.

**Reply by mano3m** [reply]
How would one do this for a Denon home device?

**Reply by OzGav** [reply]
You cannot at this time

**Reply by quadcom** [reply]
> This is available for player types that support native announcements which currently is Sonos and Sendspin. As Sendspin now has bridges to many other protocols you should find this is available for most of your players if you use Sendspin as the default protocol.

I'm running multiple esp32louder plus boards flashed with sendspin. The players work great and have been for ages. But I can't get announcements to duck and play over the music. It always pauses. Where can I get some info on how this need to be set up for it to work with a duck instead?

**Reply by mdddev** [reply]
Hi, have a look at 

https://github.com/music-assistant/support/issues/6179

The docs are a bit too vague ATM, currently AFAIK only Sonos supports this. However, for sendspin specifically, there is work being done supporting this. This is also linked in above issue.
